x/evm/keeper/msg_server.go

```diff
@@ -753,6 +753,230 @@ func (k Keeper) convertCoinToEvmBornERC20(
+		ctx,
+		evmObj,
+		evm.EVM_MODULE_ADDRESS,
+		&erc20Addr,
+		true, /*commit*/
+		contractInput,
+		Erc20GasLimitExecute,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if evmResp.Failed() {
+		return nil, fmt.Errorf("failed to burn ERC20 tokens: %s", evmResp.VmError)
+	}
+
+	// 2 | Send Bank Coins from the EVM module to the recipient
+	err = k.Bank.SendCoinsFromModuleToAccount(ctx, evm.ModuleName, toAddress, sdk.NewCoins(bankCoin))
+	if err != nil {
+		return nil, sdkioerrors.Wrap(err, "failed to send coins from module to account")
+	}
+
+	// Commit the stateDB
+	if err = stateDB.Commit(); err != nil {
+		return nil, sdkioerrors.Wrap(err, "failed to commit stateDB")
+	}
+
+	// Emit event
+	_ = ctx.EventManager().EmitTypedEvent(&evm.EventConvertEvmToCoin{
+		Sender:               sender.String(),
+		Erc20ContractAddress: erc20Addr.String(),
+		ToAddress:            toAddress.String(),
+		BankCoin:             bankCoin,
+	})
+
+	// Emit tx logs of Burn event
+	err = ctx.EventManager().EmitTypedEvent(&evm.EventTxLog{Logs: evmResp.Logs})
+	if err == nil {
+		k.updateBlockBloom(ctx, evmResp, uint64(k.EvmState.BlockTxIndex.GetOr(ctx, 0)))
+	}
+
+	return &evm.MsgConvertEvmToCoinResponse{}, nil
+}
+
+// convertEvmToCoinForERC20Originated handles conversion of ERC20 tokens that were originally ERC20
+// The EVM module doesn't own the ERC20 contract, so it transfers tokens to itself and mints bank coins
+func (k Keeper) convertEvmToCoinForERC20Originated(
+	ctx sdk.Context,
+	sender sdk.AccAddress,
+	senderEthAddr gethcommon.Address,
+	toAddress sdk.AccAddress,
+	erc20Addr gethcommon.Address,
+	amount *big.Int,
+	bankCoin sdk.Coin,
+) (*evm.MsgConvertEvmToCoinResponse, error) {
+	// needs to run first to populate the StateDB on the BankKeeperExtension
+	stateDB := k.Bank.StateDB
+	if stateDB == nil {
+		stateDB = k.NewStateDB(ctx, k.TxConfig(ctx, gethcommon.Hash{}))
+	}
+	defer func() {
+		k.Bank.StateDB = nil
+	}()
+
+	// 1 | Transfer ERC20 tokens from sender to EVM module
+	contractInput, err := embeds.SmartContract_ERC20MinterWithMetadataUpdates.ABI.Pack("transferFrom", senderEthAddr, evm.EVM_MODULE_ADDRESS, amount)
+	if err != nil {
+		return nil, err
+	}
+
+	unusedBigInt := big.NewInt(0)
+	evmMsg := core.Message{
+		To:               &erc20Addr,
+		From:             evm.EVM_MODULE_ADDRESS,
+		Nonce:            k.GetAccNonce(ctx, evm.EVM_MODULE_ADDRESS),
+		Value:            unusedBigInt,
+		GasLimit:         Erc20GasLimitExecute,
+		GasPrice:         unusedBigInt,
+		GasFeeCap:        unusedBigInt,
+		GasTipCap:        unusedBigInt,
+		Data:             contractInput,
+		AccessList:       gethcore.AccessList{},
+		BlobGasFeeCap:    &big.Int{},
+		BlobHashes:       []gethcommon.Hash{},
+		SkipNonceChecks:  true,
+		SkipFromEOACheck: true,
+	}
+
+	evmObj := k.NewEVM(ctx, evmMsg, k.GetEVMConfig(ctx), nil /*tracer*/, stateDB)
+	evmResp, err := k.CallContractWithInput(
+		ctx,
+		evmObj,
+		evm.EVM_MODULE_ADDRESS,
```

Comment:
I think we can skip the `approve` step if we use the sender's EthAddr as the `from` addr here.

Reply:
Updated in bca8095b70722a4bc40a41c571b3c7f2e9ca8cbf

    fix(evm): remove ERC20 approval requirement by switching from transferFrom to transfer
    
    This change simplifies the `ConvertEvmToCoin` flow for ERC20-originated tokens
    by replacing the use of `transferFrom` with `transfer`. As a result, token
    approvals are no longer needed prior to conversion.